ingestion-beam/src/main/java/com/mozilla/telemetry/options/SinkOptions.java

```diff
@@ -66,6 +66,33 @@
 
   void setSchemaAliasesLocation(ValueProvider<String> value);
 
+  @Description("Method of reading from BigQuery; the table will either be exported to GCS"
+      + " (GA and free, but may take some time to export and may hit quotas) or accessed using the "
+      + " BigQuery Storage API (beta and some cost, but faster and no quotas)")
+  @Default.Enum("export")
+  BigQueryReadMethod getBqReadMethod();
+
+  void setBqReadMethod(BigQueryReadMethod value);
+
+  @Description("When --bqReadMethod=storageapi, all rows of the input table are read by default,"
+      + " but this option can take a SQL text filtering statement, similar to a WHERE clause;"
+      + " currently, only a single predicate that is a comparison between a column and a constant"
+      + " value is supported; a likely choice to limit partitions would be something like"
+      + " \"CAST(submission_timestamp AS DATE)"
+      + " BETWEEN CAST('2020-01-10' AS DATE) AND CAST('2020-01-14' AS DATE)\"; see"
+      + " https://cloud.google.com/bigquery/docs/reference/storage/rpc/google.cloud.bigquery.storage.v1beta1#tablereadoptions")
+  String getBqRowRestriction();
+
+  void setBqRowRestriction(String value);
+
+  @Description("When --bqReadMethod=storageapi, all fields of the input table are read by default,"
+      + " but this option can take a comma-separated list of field names, in which case only the"
+      + " listed fields will be read, saving costs; when reading decoded payload_bytes, none of the"
+      + " metadata fields are needed, so setting --bqSelectedFields=payload is recommended")
+  List<String> getBqSelectedFields();
+
+  void setBqSelectedFields(List<String> value);
```

Comment:
Note that all three of these new options are compile-time only. I suspect that the latter two could be specified at runtime, but no `ValueProvider` interface exists. We could consider filing an upstream PR; see #736